internal/config/config.go

```diff
@@ -215,7 +215,10 @@ type Instrumentation struct {
 
 // SmartRetry represents the settings for retry strategy.
 type SmartRetry struct {
+	// FailedClassesOnly retries failed classes for Espresso and XCUITest
 	FailedClassesOnly bool `yaml:"failedClassesOnly" json:"-"`
+	// FailedTestsOnly retries only failed tests or failed spec files for Cypress/Playwright/Cucumber/TestCafe
```

Comment:
Full sentences should end with a dot 🙏 

```suggestion
	// FailedClassesOnly retries failed classes for Espresso and XCUITest.
	FailedClassesOnly bool `yaml:"failedClassesOnly" json:"-"`
	// FailedTestsOnly retries only failed tests or failed spec files for Cypress/Playwright/Cucumber/TestCafe.
```

1. Shouldn't this documentation also go into the API spec? The user won't see this here 😅 Can you tell, as a user, which you should be using when?
2. So `FailedTestsOnly` retries either a test case or spec files. How does the user know which is done when? This also makes the API inconsistent. `FailedClassesOnly` --> retries failed classes. `FailedTestsOnly` --> retries either test cases or entire spec files. The former is more concrete, the latter is 🤷 , because it depends. Then why not have just one `FailedTestsOnly`, where "it depends". Or concrete ones only: FailedClassesOnly, FailedSpecOnly, FailedTestCasesOnly. We need to give this more thought, because it seems to be growing organically right now.

Reply:
After thinking about this, I just used `FailedOnly` to present this testing strategy. This applies to all of our frameworks and can be treated differently per each framework.

Reply:
Update: I just keep `FailedClassesOnly` but marked it as deprecated. Users can still use it.